src/Extensions/Services.Remoting/Internal/Client/ServiceRemotingClientWrapper.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Fabric;
+using System.Threading.Tasks;
+using Microsoft.Omex.Extensions.Abstractions.Activities;
+using Microsoft.ServiceFabric.Services.Remoting.V2;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
+
+namespace Microsoft.Omex.Extensions.Services.Remoting.Client
+{
+	/// <summary>
+	/// Wrapper for adding Omex headers to Remoting request
+	/// </summary>
+	internal class ServiceRemotingClientWrapper : IServiceRemotingClient
+	{
+		/// <summary>
+		/// Suffix for this class activities
+		/// </summary>
+		/// <remarks>
+		/// Should end with RequestOut to be enabled by our telemetry
+		/// </remarks>
+		private const string RequestOutActivitySuffix = "RequestOut";
+
+		private const string OneWayMessageActivityName = Diagnostics.DiagnosticListenerName + "OneWay" + RequestOutActivitySuffix;
+
+		private const string RequestActivityName = Diagnostics.DiagnosticListenerName + RequestOutActivitySuffix;
+
+		private readonly DiagnosticListener m_diagnosticListener;
+
+		internal readonly IServiceRemotingClient Client;
+
+		public ServiceRemotingClientWrapper(IServiceRemotingClient client, DiagnosticListener? diagnosticListener = null)
+		{
+			Client = client;
+			m_diagnosticListener = diagnosticListener ?? Diagnostics.DefaultListener;
+		}
+
+		public ResolvedServicePartition ResolvedServicePartition
+		{
+			get => Client.ResolvedServicePartition;
+			set => Client.ResolvedServicePartition = value;
+		}
+
+		public string ListenerName
+		{
+			get => Client.ListenerName;
+			set => Client.ListenerName = value;
+		}
+
+		public ResolvedServiceEndpoint Endpoint
+		{
+			get => Client.Endpoint;
+			set => Client.Endpoint = value;
+		}
+
+		public async Task<IServiceRemotingResponseMessage> RequestResponseAsync(IServiceRemotingRequestMessage requestMessage)
+		{
+			Activity? activity = m_diagnosticListener.CreateAndStartActivity(OneWayMessageActivityName);
+			requestMessage.AttachActivityToOutgoingRequest(activity);
+			IServiceRemotingResponseMessage? responseMessage = null;
+
+			try
+			{
+				responseMessage = await Client.RequestResponseAsync(requestMessage).ConfigureAwait(false);
+				activity?.SetResult(TimedScopeResult.Success);
+			}
+			catch (Exception ex)
+			{
+				m_diagnosticListener.ReportException(ex);
+				throw;
+			}
+			finally
+			{
+				m_diagnosticListener.StopActivityIfExist(activity);
+			}
+
+			return responseMessage;
+		}
+
+		public void SendOneWay(IServiceRemotingRequestMessage requestMessage)
+		{
+			Activity? activity = m_diagnosticListener.CreateAndStartActivity(RequestActivityName);
+			requestMessage.AttachActivityToOutgoingRequest(activity);
+
+			try
+			{
+				Client.SendOneWay(requestMessage);
+				activity?.SetResult(TimedScopeResult.Success);
+			}
+			catch (Exception ex)
```

Comment:
when !(ex is OutOfMemoryException)

We should never try to catch and log this exception type, as the act of logging it could cause a separate OutOfMemoryException to be thrown, thus hiding the original exception and stack trace.

Reply:
good point, I'll add an exclusions

Reply:
OutOfMemory is more likely to mean "don't have enough memory for what you want" than "don't have any memory left", so logger is likely to succeed, but see the docs for suggestions on how to handle that exception: https://docs.microsoft.com/en-us/dotnet/api/system.outofmemoryexception?view=netframework-4.8